plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/SetPropertyAction.java

```diff
@@ -102,6 +102,12 @@ public int execute(Action action, Event event) {
             propertyValue = format.format(date);
         }
 
+        if (propertyValue != null && propertyValue.equals("eventArrivalTime")) {
```

Comment:
I would rename "eventArrivalTime" to "timeStamp" or "actionExecutionTimeStamp".
Because it's not really true that this timeStamp is the real event arrival time. Event could arrive sooner and the action being executed few seconds after.